onnxruntime/core/providers/cuda/cu_inc/common.cuh

```diff
@@ -119,6 +119,18 @@ __device__ __inline__ double _Erf(double a) { return erf(a); }
 template <>
 __device__ __inline__ half _Erf(half a) { return half(erff((float)a)); }
 
+template <typename T>
+__device__ __inline__ T _Round(T a);
+
+template <>
+__device__ __inline__ float _Round(float a) { return rintf(a); }
+
+template <>
+__device__ __inline__ double _Round(double a) { return rint(a); }
+
+template <>
+__device__ __inline__ half _Round(half a) { return half(rintf((float)a)); }
```

Comment:
For half, it is possible to use half2 to optimize if input element count is divisible by 2. 

There is round function for half2: h2rint

Reply:
I ll read about this API and see if it can be used. Thanks.

Reply:
Any specific reason not to just stick with what we are doing for other half operations ? Everything else casts it to float, calls the float method, and converts the result back to half before returning....

Reply:
I think a half round function could be faster than float32 round plus two extra conversions:
   use half function:  half + 0.5 => "int"  (set 11 fraction bits to zero).
   use float function:  convert float16 to float32; float + 0.5 => "int" (set fraction bits to zero)  => convert float 32 to float 16.
Only drawback is half function might not be avaiable on cuda architecture < 5.3. So, for older cuda arch has to use the other way.

This also applies to other half operations. We shall try use native half functions when available.